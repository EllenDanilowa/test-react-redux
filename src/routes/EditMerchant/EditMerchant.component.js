import React from 'react';
import PropTypes from 'prop-types';
import MerchantForm from '../../components/MerchantForm/MerchantForm';

const EditMerchant = ({merchant, updateMerchant}) => {
  const update = (item) => updateMerchant(item);

  return (
    <div>
      <MerchantForm merchant={merchant}
                    submit={update}
                    title="Edit a merchant"
                    submitTitle="Update"/>
    </div>
  );
};

EditMerchant.propTypes = {
  updateMerchant: PropTypes.func.isRequired,
  merchant: PropTypes.object,
};

export default EditMerchant;
