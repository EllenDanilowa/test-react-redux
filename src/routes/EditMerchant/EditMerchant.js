import React, {Component} from 'react';
import PropTypes from 'prop-types';
import MerchantForm from '../../components/MerchantForm/MerchantForm';
import {editMerchant} from "../../redux/actions/merchant";
import {connect} from "react-redux";

class EditMerchant extends Component {
  constructor(props) {
    super(props);

    //const id = this.props.match.params.id;
    this.item = {};
    this.editMerchant = this.editMerchant.bind(this);
  }

  editMerchant(merchant) {
    this.props.editMerchant(merchant);
  }

  render() {
    return (
      <div>
        <h3>Edit a merchant</h3>
        <MerchantForm initValue={this.item}/>
      </div>
    );
  }
}

EditMerchant.propTypes = {
  editMerchant: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  editMerchant: (merchant) => {
    dispatch(editMerchant(merchant));
  }
});

export default connect(() => ({}), mapDispatchToProps)(EditMerchant);
