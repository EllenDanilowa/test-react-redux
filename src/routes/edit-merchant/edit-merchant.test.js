import React from 'react';
import renderer from 'react-test-renderer';
import EditMerchant from './edit-merchant.component';

jest.mock('../../components/merchant-form/merchant-form', () => () => 'Merchant Form component');

describe('EditMerchant', () => {
  let element;
  let merchant;
  let updateMerchant;

  const createElement = (props) => {
    const template = (<EditMerchant {...props}/>);

    return renderer.create(template).toJSON();
  };

  beforeEach(() => {
    merchant = {};
    updateMerchant = jest.fn();
  });

  it('renders component correctly', () => {
    element = createElement({merchant, updateMerchant});

    expect(element).toMatchSnapshot();
  });
});
