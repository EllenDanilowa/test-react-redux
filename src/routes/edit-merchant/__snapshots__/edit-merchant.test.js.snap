// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMerchant renders component correctly 1`] = `
<div
  className="sc-AxirZ cSToxy"
>
  <h3
    className="sc-AxjAm hnoMwN"
  >
    Edit a merchant
  </h3>
  Merchant Form component
</div>
`;
